rotkehlchen/history/events/structures/asset_movement.py

```diff
@@ -240,10 +241,13 @@ def create_asset_movement_with_fee(
         asset=asset,
         balance=Balance(amount),
         unique_id=unique_id,
+        identifier=identifier,
         extra_data=extra_data,
     )]
     if fee != ZERO:
         events.append(AssetMovement(
+            # we can increment here since it's added after the main event.
+            identifier=identifier + 1 if identifier is not None else None,
```

Comment:
This is very hacky. If at any point for whatever reason we somehow context switch between the creation of the main asset movement and the asset movement fee they would end up getting non-consecutive identifiers. Making this assumptions is dangerous.

Reply:
so i went with the approach of passing the `database` to the schema and querying the `identifier` from the db instead of making such an assumption.